Mage.Tests/src/test/java/org/mage/test/cards/copy/CleverImpersonatorTest.java

```diff
@@ -181,9 +186,18 @@ Whenever enchanted land is tapped for mana, its controller adds two mana in any
         addCard(Zone.BATTLEFIELD, playerA, "Island", 6);
         addCard(Zone.BATTLEFIELD, playerA, "Forest", 6);
 
-        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, dReflection, "Forest"); // enchant a forest
+        setStrictChooseMode(true);
+
+        // TODO: Something is broken with this test. Both of these choices should be using addTarget.
+        //       The second one uses setChoice for some reason
+        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, dReflection); // enchant a forest
+        addTarget(playerA, "Forest");
+        setChoice(playerA, "Blue");
+        setChoice(playerA, "Blue");
         castSpell(1, PhaseStep.POSTCOMBAT_MAIN, playerA, impersonator);
+        setChoice(playerA, true); // Use Clever Impersonator's ability
         setChoice(playerA, dReflection); // have Impersonator enter as copy of Dawn's Reflection
+        setChoice(playerA, "Island");
```

Comment:
This seems like a bug in the code?

Should both `addTarget(playerA, "Forest");` and `setChoice(playerA, "Island");` use the an addTarget call? They both require targeting. Changing the `setChoice(playerA, "Island");` to a `addTarget(playerA, "Island");` will not run.

Reply:
There were some wrong usages in choose logic (example: game engine calls `choose()` instead `chooseTarget()`) sometime, so it's ok to setup targets as setChoice/addTarget in some abilities. Use strict choose mode, it will show miss command (choice or target).

You are commented standard choices in the test:
![shot_221004_082745](https://user-images.githubusercontent.com/8344157/193734433-9700caf8-1d3a-47cc-9c83-edad4e6e3156.png)

Why?

It's work fine with that changes:
![shot_221004_083003](https://user-images.githubusercontent.com/8344157/193734553-fdc57013-8ccb-4547-9723-78b5dead84d6.png)

Reply:
I believe I had it commented out by mistake. With it commented out, the targets will be autochosen. I will re-enable the manual targeting.

Reply:
> There were some wrong usages in choose logic (example: game engine calls choose() instead chooseTarget()) sometime, so it's ok to setup targets as setChoice/addTarget in some abilities. Use strict choose mode, it will show miss command (choice or target).

In this came this seems like a bug since both of them should be using `addTarget`. I am worried that since the copied Dawn's Reflection uses `setChoice` that it's not actually targeting.